packages/legend-query-builder/src/components/QueryBuilder.tsx

```diff
@@ -367,10 +370,35 @@ export const QueryBuilder = observer(
       }
       return null;
     };
+
+    const undo = (): void => {
+      queryBuilderState.changeHistoryState.undo();
+    };
+
+    const redo = (): void => {
+      if (queryBuilderState.isUndoRedoUnderContext) {
+        queryBuilderState.changeHistoryState.redo();
+      }
+    };
+
+    useEffect(() => {
+      // this condition is for passing all exisitng tests because when we initialize a queryBuilderState for a test,
+      // we use an empty RawLambda with an empty class and this useEffect is called earlier than initializeWithQuery()
+      if (
+        (queryBuilderState.isQuerySupported && queryBuilderState.class) ||
+        !queryBuilderState.isQuerySupported
```

Comment:
Is it this condition you are adding to prevent test failures `!queryBuilderState.isQuerySupported` ? What error do the tests throw?

Reply:
all existing tests will fail to render result for query builder

```
  const renderResult = render(
    <ApplicationStoreProvider store={MOCK__applicationStore}>
      <TEST__BrowserEnvironmentProvider>
        <ApplicationFrameworkProvider>
          <QueryBuilder queryBuilderState={queryBuilderState} />
        </ApplicationFrameworkProvider>
      </TEST__BrowserEnvironmentProvider>
    </ApplicationStoreProvider>,
  );
```

 as this useEffect is called earlier and it will trigger queryBuilderState.buildQuery()
and will eventually fail at ..../src/stores/QueryBuilderValueSpecificationBuilder.ts line 158
```
  const _class = guaranteeNonNullable(
    queryBuilderState.class,
    'Class is required to build query',
  );
```

🫣🫣🫣🫣